### PR TITLE
Remote configuration for Call screen

### PR DESCRIPTION
**Pull request by AndriiHorishniiMOC**
Allow using remote UI configuration on the Call screen.

[MOB-1570](https://glia.atlassian.net/browse/MOB-1570)

Implemented:
```
RemoteConfiguration:
    - callScreen
        - background
        - bottomText
        - duration
        - operator
        - topText
        - buttonBar
            - chatButton, minimizeButton, muteButton, speakerButton, videoButton
                - inactive, active, selected
                    - background
                    - imageColor
                    - title
```

### CODE REVIEW

**Comment by gugalo** [comment]
Nice screenshots!! 👍

File: widgetssdk/src/main/java/com/glia/widgets/UiTheme.java [hunk 1]
```diff
@@ -711,8 +620,19 @@ public void setIconUpgradeAudioDialog(@DrawableRes Integer iconUpgradeAudioDialo
             this.iconUpgradeAudioDialog = iconUpgradeAudioDialog;
         }
 
+        @Deprecated
```
Comment (gugalo) [inline widgetssdk/src/main/java/com/glia/widgets/UiTheme.java:623]: Sorry if it is obvious, but I don't understand why only some methods related to styling are deprecated. For example, why `setIconCallChat` method is deprecated and `setIconChatAudioUpgrade` is not. Or is it just that they are not related to the call screen?

File: widgetssdk/src/main/java/com/glia/widgets/call/CallView.java [hunk 2]
```diff
@@ -644,18 +652,26 @@ private void showScreenSharingDialog() {
...
                 notActivatedContentDescription
         ));
+        floatingActionButton.setEnabled(isEnabled);
+        setBarButtonLabelAppearance(
+                textView,
+                isActivated ?
+                        StyleUtils.FabButtonStatus.SELECTED :
+                        isEnabled ?
+                                StyleUtils.FabButtonStatus.ACTIVE :
+                                StyleUtils.FabButtonStatus.INACTIVE
```
Comment (gugalo) [inline widgetssdk/src/main/java/com/glia/widgets/call/CallView.java:673]: I don't think this is reasonable or readable to pass as a second argument a double-inline-if statement )
Can you please try to extract it somehow?

File: widgetssdk/src/main/java/com/glia/widgets/chat/ChatActivity.java [hunk 3]
```diff
@@ -140,6 +140,7 @@ private Configuration.Builder getConfigurationBuilder() {
 
     private void navigateToSurvey(UiTheme theme, Survey survey) {
         Intent newIntent = new Intent(getApplicationContext(), SurveyActivity.class);
+//        newIntent.putExtra(GliaWidgets.REMOTE_CONFIGURATION, ); // TODO
```
Comment (gugalo) [inline widgetssdk/src/main/java/com/glia/widgets/chat/ChatActivity.java:143]: Probably left-overs?

File: widgetssdk/src/main/java/com/glia/widgets/filepreview/ui/FilePreviewActivity.java [hunk 4]
```diff
@@ -74,6 +74,7 @@ public static Intent intent(Context context, String itemId, String itemName, UiT
         Intent intent = new Intent(context, FilePreviewActivity.class);
         intent.putExtra(IMAGE_ID_KEY, itemId);
         intent.putExtra(IMAGE_ID_NAME, itemName);
+//        intent.putExtra(GliaWidgets.REMOTE_CONFIGURATION, ); // TODO
```
Comment (gugalo) [inline widgetssdk/src/main/java/com/glia/widgets/filepreview/ui/FilePreviewActivity.java:77]: Probably left-overs?

File: widgetssdk/src/main/java/com/glia/widgets/view/configuration/ButtonConfiguration.java [hunk 5]
```diff
@@ -133,9 +133,11 @@ public Builder textConfiguration(TextConfiguration textConfiguration) {
         }
 
         public ButtonConfiguration build() {
+            // TODO: builder should not set default values
```
Comment (gugalo) [inline widgetssdk/src/main/java/com/glia/widgets/view/configuration/ButtonConfiguration.java:136]: TODO

File: widgetssdk/src/main/java/com/glia/widgets/view/configuration/LayerConfiguration.java [hunk 6]
```diff
@@ -4,59 +4,116 @@
...
+        return backgroundColor.getColor();
+    }
+
+    @Nullable
+    public ColorConfiguration getBorderColorConfiguration() {
         return borderColor;
     }
 
-    public int getBorderWidth() {
+    @Deprecated // Remove after refactoring
```
Comment (gugalo) [inline widgetssdk/src/main/java/com/glia/widgets/view/configuration/LayerConfiguration.java:52]: Should be removed?

**Review by gugalo — CHANGES_REQUESTED** [review]
If I remember correctly there have been discussions regarding following the right format for the commit messages. 
There are some guides [here](https://glia.atlassian.net/wiki/spaces/ENG/pages/15728856/Style+Guide+Conventions#Git-Commit-Messages)
But if shortly:
- Add a few words to provide the context of this commit
- Mark any unfinished or important parts
- Add a ticket code at the end eg: MOB-1570

For an example see one line commit from Gerson: 
https://github.com/salemove/ios-sdk/commit/b6681020ccaf8d63ad92f4a573283aa36355a9a0

**Comment by gugalo** [comment]
Please try to divide it into smaller PRs next time.
But it is very great work. My praises @AndriiHorishniiMOC!

File: app/src/main/res/values/setup.xml [hunk 7]
```diff
@@ -1,4 +1,4 @@
 <?xml version="1.0" encoding="utf-8"?>
 <resources xmlns:tools="http://schemas.android.com/tools" tools:ignore="ExtraTranslation,TypographyDashes,Typos">
-    <string name="default_queue_id">a0b87175-4ed1-44a7-9e78-e798f45d233f</string>
+    <string name="default_queue_id">f2e6ca5f-2161-4298-8774-5a847f3d6d7e</string>
```
[the remark this answers is not in the document]
Comment (AndriiHorishniiMOC) [reply app/src/main/res/values/setup.xml]: Oops. You saw nothing :)

File: widgetssdk/src/main/java/com/glia/widgets/helper/Utils.java [hunk 8]
```diff
@@ -508,6 +566,13 @@ private static <T> T getConfiguration(
         return newConf != null ? newConf : oldConf;
     }
 
+    public static UiTheme getFullHybridTheme(RemoteConfiguration remoteConfiguration, UiTheme newTheme, UiTheme oldTheme) {
+        UiTheme defaultUiTheme = UiTheme.getDefaultUiTheme();
+        UiTheme fullHybridOldTheme = getFullHybridTheme(oldTheme, defaultUiTheme);
+        UiTheme theme = getFullHybridTheme(newTheme, fullHybridOldTheme);
+        return ThemeUnits.apply(theme, remoteConfiguration);
+    }
```
Comment (andrews-moc) [inline widgetssdk/src/main/java/com/glia/widgets/helper/Utils.java:574]: It's a bit confusing to me that `getFullHybridTheme()` is called twice. And I think there should be better names instead of `newTheme` and `oldTheme`.

**Comment by andrews-moc** [comment]
@AndriiHorishniiMOC You've really done a good deal of work. 👍

**Review by DavDo — APPROVED** [review]
